src/views/Generator/TestOptions/Thresholds/ThresholdRow.tsx

```diff
@@ -30,12 +31,26 @@ export function ThresholdRow({ field, index, remove }: ThresholdRowProps) {
     formState: { errors },
     control,
     watch,
+    setValue,
   } = useFormContext<ThresholdData>()
 
   const urlOptions = useThresholdURLOptions()
-  const threshold = watch('thresholds')[index]
+  const threshold = watch('thresholds')[index] as Threshold
   const theme = useTheme()
 
+  // Handle selected statistic when the metric field changes
+  useEffect(() => {
```

Comment:
The available options in the 'Statistic' field depend on the selection made in the 'Metric' field. Here we handle the case when a different option is chosen in the 'Metric' field that is incompatible with the current selected 'Statistic'.